sdk/core/scheduler/common.h

```diff
@@ -247,6 +247,10 @@ namespace
 		HeapObject(struct SObjStruct *heapCapability, T *allocatedObject)
 		  : pointer(allocatedObject, heapCapability)
 		{
+			if (__builtin_cheri_tag_get(allocatedObject) == false)
```

Comment:
Why not just ! __builtin_cheri_tag_get(allocatedObject)?

Reply:
No particular reason, I can update. I do not have a strong preference.